packages/terra-aggregator/src/Aggregator.jsx

```diff
@@ -0,0 +1,229 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  /**
+   * The components that will be managed by the Aggregator. Each component provided will be provided with an 'aggregatorDelegate' prop
+   * containing the Aggregator API. Keys must also be provided to allow the Aggregator to uniquely identify the component.
+   * Immutable objects are preferred.
+   */
+  items: PropTypes.arrayOf(PropTypes.shape({
+    key: PropTypes.string.isRequired,
+    component: PropTypes.element.isRequired,
+  })),
+  /**
+   * A function that will override the Aggregators default render. The function will receive an Object parameter data necessary for the
+   * render process.
+   */
+  render: PropTypes.func,
+  /**
+   * A function that will be provided to Aggregator items that have received focus. The function must adhere to the standardized
+   * AppDelegate disclosure API.
+   */
+  disclose: PropTypes.func,
+};
+
+const defaultProps = {
+  items: [],
+};
+
+class Aggregator extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.requestFocus = this.requestFocus.bind(this);
+    this.releaseFocus = this.releaseFocus.bind(this);
+    this.setFocusState = this.setFocusState.bind(this);
+    this.resetFocusState = this.resetFocusState.bind(this);
+    this.renderItems = this.renderItems.bind(this);
+
+    this.state = {
+      focusedItemId: undefined,
+      focusedItemState: undefined,
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { items } = this.props;
+    const { focusedItemId } = this.state;
+
+    if (nextProps.items !== this.props.items) {
+      // If the currently focused item is not present in the new items set,
+      // the focus is forcefully released to clean up any presented disclosures.
+      let focusItemIsPresent;
+      items.forEach((item) => {
+        if (item.key === focusedItemId) {
+          focusItemIsPresent = true;
+        }
+      });
+
+      if (!focusItemIsPresent) {
+        this.releaseFocus(undefined, true);
```

Comment:
Is this line necessary?

Reply:
Yes. If the Aggregator item that holds focus is removed from the Aggregator on a subsequent render, we need to ensure the Aggregator's internal state is updated. That's all this is doing. It probably won't happen very often.